### PR TITLE
[Answer Query Using Materialized Views] Correct PlannerInfo fields after rewritten.

### PR DESCRIPTION
**Pull request by avamingli**
If we succeed to replace SQL with materialized view, correct PlannerInfo's stuff to new one like:
simple_rte_array, simple_rel_array and etc.
They may be used to fetch rels after planner, for
example: use physical rel's target list in create_plan().

Authored-by: Zhang Mingli avamingli@gmail.com

fix #ISSUE_Number

---

### Change logs

_Describe your change clearly, including what problem is being solved or what feature is being added._

_If it has some breaking backward or forward compatibility, please clary._

### Why are the changes needed?

_Describe why the changes are necessary._

### Does this PR introduce any user-facing change?

_If yes, please clarify the previous behavior and the change this PR proposes._

### How was this patch tested?

_Please detail how the changes were tested, including manual tests and any relevant unit or integration tests._

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [x] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution(*One-time setup*).
- [ ] Learn the [coding contribution guide](https://cloudberrydb.org/contribute/code), including our code conventions, workflow and more.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [x] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to request `cloudberrydb/dev` team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by avamingli** [comment]
Fix #344

File: src/backend/optimizer/plan/aqumv.c [hunk 1]
```diff
@@ -422,6 +421,12 @@ answer_query_using_materialized_views(PlannerInfo *root,
 		{
 			root->parse = mvQuery;
 			root->processed_tlist = subroot->processed_tlist;
+			root->agginfos = subroot->agginfos;
+			root->aggtransinfos =  subroot->aggtransinfos;
```
Comment (yjhjstz) [inline src/backend/optimizer/plan/aqumv.c:425]: hasNonCombine need assign ?

Comment (avamingli) [reply src/backend/optimizer/plan/aqumv.c:425]: Good suggestion, I will have a look.

Comment (avamingli) [reply src/backend/optimizer/plan/aqumv.c:428]: Fixed, we need to correct them, as well as another three related to agg.
Though, some of them may not be used due to IVM and AQUMV process, we keep them right.
Thanks for reviewing.

**Review by yjhjstz — APPROVED** [review]

**Comment by avamingli** [comment]
Pushed, thanks for reviewing.

**Pull request merged**